miniml/compiler/compile.scm

```diff
@@ -932,19 +962,30 @@
     (if (pair? r) r (errorp "Not found in env: " name))
     ))
 
+(define-immutable-record-type <functor>
+  (mkfunctor env argname body)
+  functor?
+  (env functor-get-env functor-with-env)
+  (argname functor-get-argname functor-with-argname)
+  (body functor-get-body functor-with-body))
+
+(define (module-get-env mod)
+  (assert (equal? (car mod) 'VModule))
+  (car (cdr mod)))
```

Comment:
same here, pattern-matching with `match-let` would result in more readable failure messages than `assert`